README.md

```diff
@@ -1896,9 +1898,12 @@ workbook.xlsx.readFile(filename)
     // use workbook
   });
 
-// pipe from stream
+// read from stream
 var workbook = new Excel.Workbook();
-stream.pipe(workbook.xlsx.createInputStream());
```

Comment:
No, it's been removed because I removed `.createInputStream()`.  The reason is that a) it can be completely replaced by `.read()` and b) it uses a lot of non-standard, custom code and behaves in really unexpected ways.

Since there is a replacement for it (`.read()`), we don't lose any functionality, but we make it   easier to maintain the library. However, some consumers of our library will have to make (simple) changes in order to use `4.0.0`, so is it worth it? Let's talk about the second point in detail:

It looks nice that we can use `stream.pipe(createInputStream())`, but if you look at the implementation if `createInputStream`, you start to realize that it's not a stream at all (https://github.com/exceljs/exceljs/pull/1135/files#diff-bae1d9f34b0ea7278956352e70c79c5e). It's a super custom implementation based on event emitters, has nothing do with actual writeable streams and re-implements a lot of logic itself. The code itself even says:
```
// The purpose of this module is to wrap the js-zip library into a streaming zip library
// since most of the exceljs code uses streams.
// One day I might find (or build) a properly streaming browser safe zip lib
```
The good thing is that with generators, we can actually build something that is standard, simple and easy to maintain, like I did here: https://github.com/exceljs/exceljs/pull/1135/files#diff-582ff38cb0a89a7a808debcaf9965d00R260

Now that the internals have been drastically simplified using the simpler async iteration control flow, it's actually difficult to re-expose a function that behaves just like `createInputStream` - and coupled with the fact that you could just use `.read()`, I think we should drop it.

Reply:
👍 for getting rid of flow control

Reply:
Well, I think that for version 3.* we should mark this function as `OBSOLETE`/`DEPRECETED` first with message explain when it will be removed (from v4 -may 2020 ) and how to upgrade (link to migration instruction - now we haven't even instruction for that).

I little worry that exceljs users may have headache after upgrade to v4 and finally stay in v3  

I see what awesome work you did and how it's helpful for using xlsx files. I want to push the merge button ASAP. These changes will make a big step to the future 😄 

So, remain, I think we should:
1. Obsolete this function in v3.9.3 (it hasn't published yet :/ )
2. Add any instruction what exactly should do after upgrade to v4.
3. Schedule time when we merge and publish.


orr.. make some adapter for `createInputStream` (if possible), and obsolete (or not) in v4.

Reply:
I think that's a great idea - I'll create a PR deprecating it in 3.9.3. How should we mark it? Using `console.log` each time it gets called?

Reply:
Done, using `console.warn`: https://github.com/exceljs/exceljs/pull/1209

Reply:
https://jsdoc.app/tags-deprecated.html